addons/AnnounceTarget/AnnounceTarget.lua

```diff
@@ -0,0 +1,112 @@
+defaults.AutoAnnounce = false
+ 
+settings = config.load(defaults)
+ 
+adherents_map = {['Steadfast Adherent']="PLD, DEF+", ['Furtive Adherent']="WHM, MDB+", ['Occult Adherent']="WAR, EVA+",
+        ['Fleet Adherent']="WAR, Haste+", ['Brawny Adherent']="DRK, ATK+", ['Martial Adherent']="DRK,Regain+",
+        ['Honed Adherent']="RDM, Fast Cast+", ['Insidious Adherent']="RDM, MEVA+", ['Hexbreaking Adherent']="BLM, MAB+"}
+chatmodes = S{'say','party','linkshell','linkshell2','shout','echo','s','p','l','l2','sh'}
+false_values = S{'false','off','f','0'}
+true_values = S{'true','on','t','1'}
+moblist = S{}
+mob = windower.ffxi.get_mob_by_target('st') or windower.ffxi.get_mob_by_target('t')
+        
+windower.register_event('addon command', function (command,...)
+    command = command and command:lower() or 'help'
+    local args = T{...}
+    if command == 'reload' then
+        windower.send_command('lua reload AnnounceTarget')
+    elseif command == 'unload' then
+        windower.send_command('lua unload AnnounceTarget')
+    elseif command == 'chatmode' or command == 'cm' then
+        if args[1] ~= nil and chatmodes:contains(args[1]) then
+            log(' ***** Chat Mode changed to "'..args[1]..'" *****')
+            settings.AnnounceMode = args[1]
+            config.save(settings)
+        else
+            error(' ***** That is not a valid chat mode *****')
+        end
+    elseif command == 'announce' or command == 'a' then
+        announce(mob.name)
+    elseif command == 'autoannounce' or command == 'aa' then
+        local value = args[1] and args[1]:lower() or nil
+        if not value then
+            settings.AutoAnnounce = not settings.AutoAnnounce
+        elseif false_values:contains(value) or true_values:contains(value) then
+            settings.AutoAnnounce = not false_values:contains(args[1]:lower())
+        else
+            error(' ***** "'..args[1]..'" is not a valid setting for AutoAnnounce *****')
+            return
+        end
+        log(' ***** AutoAnnounce changed to "',settings.AutoAnnounce,'" *****')
+        config.save(settings)
+    elseif command == 'clear' or command == 'c' then
+        moblist:clear()
+		log(' ***** Previously announced targets table cleared *****')
+    elseif command == 'help' then
+        log(' *** '.._addon.name..' v'.._addon.version..' - Authors: '.._addon.author..' ***')
+        log(' help -> Displays this message')
+        log(' chatmode -> Changes chat output mode. Available settings: say/party/linkshell/linkshell2/shout/echo')
+        log(' autoannounce -> Turns AutoAnnounce on or off. Accepted settings: on/true/false/off')
+        log(' announce -> Manually announces for the current target')
+        log(' clear -> Clears the list of announced mobs during AutoAnnounce mode on')
+    end
+end)
+
+function announce(name)
+    if adherents_map[name] then
+        windower.send_command('input /'..settings.AnnounceMode..' '..name..' buff is ==> '..adherents_map[name])
```

Comment:
Just a note, I'll be adding `windower.input` to the next version of LuaCore, so you don't need to do `windower.send_command('input ...')` anymore.